rmgpy/kinetics/diffusionLimited.py

```diff
@@ -115,18 +114,43 @@ def getDiffusionLimit(self, T, reaction, forward=True):
             reacting = reaction.reactants
         else:
             reacting = reaction.products
-        assert len(reacting)==2, "Can only calculate diffusion limit in a bimolecular direction"
+
+        if len(reacting) < 2:
+            raise Exception("Cannot calculate diffusion limit for a unimolecular reaction")
+
         radii = 0.0
-        diffusivities = 0.0
+        diffusivities = []
         for spec in reacting:
             soluteData = self.database.getSoluteData(spec)
             # calculate radius with the McGowan volume and assuming sphere
             radius = ((75 * soluteData.V / constants.pi / constants.Na) ** (1. / 3)) / 100  # m
             diff = soluteData.getStokesDiffusivity(T, self.getSolventViscosity(T))
-            radii += radius  # meters
-            diffusivities += diff #m^2/s
+            radii += radius  # m
+            diffusivities.append(diff)  # m^2/s
+
+        # Calculate Smoluchowski kinetics for any reaction order
+        # Flegg, SIAM J. Appl. Math., 76 (4), 2016
+        N = len(reacting)
+
+        Dinv = 1.0 / np.array(diffusivities)
+        Dhat = np.empty(N-1)
+
+        alpha = (3.0*N - 5.0) / 2.0
+        denom = 0.0
+
+        for i in range(N-1):
+            Dbar = 1.0 / np.sum(Dinv[:(i+1)])
+            Dhat[i] = diffusivities[i+1] + Dbar
+
+            for j in range(i+1, N):
+                denom += 1.0 / (diffusivities[i]*diffusivities[j])
+
+        delta = np.sum(Dinv) / denom
+        k_diff = (np.prod(Dhat)**1.5
+                  * 4.0*constants.pi**(alpha+1.0)/math.gamma(alpha)
+                  * (radii/np.sqrt(delta))**(2.0*alpha)
+                  * constants.Na**(N-1.0))  # m^(3*(N-1))/mol^(N-1)/s
```

Comment:
So this entirely replaces the original calculation even for the 2 reactant case?

Reply:
Yes. It simplifies exactly to the previous expression for 2 reactants.